js/main.js

```diff
@@ -1,45 +1,491 @@
+      Object.values(playerShips)[shipCounter].name
+    } ${Object.values(playerShips)[shipCounter].length} Tiles `;
+  } else if (winner !== null) {
+    playAgainBtn.innerHTML = "Play Again";
+    activePlayerEl.innerHTML = `${winner === 1 ? "PLAYER WINS!" : "CPU WINS!"}`;
+    document.body.appendChild(playAgainBtn);
+    return;
+  } else {
+    verticalLabelEl.style.display = "none";
+    activePlayerEl.innerHTML = `${turn === 1 ? "PLAYER'S TURN" : "CPU'S turn"}`;
+  }
+  return sleep(700);
+}
+
+//Checking if the ship can be placed on the selected div
+function canBePlaced(board, rows, columns, vertOrHoriz, ship) {
+  if (vertOrHoriz === "horizontal") {
+    for (let i = columns; i < columns + ship.length; i++) {
+      if (i >= LENGTH || board[rows][i] !== 0) return false;
+    }
+  } else {
+    for (let i = rows; i < rows + ship.length; i++) {
+      if (i >= LENGTH || board[i][columns] !== 0) return false;
+    }
+  }
+  return true;
+}
+
+//Actually placing the ships after confirming that they can be placed
+function placeShips(board, rows, columns, vertOrHoriz, ship) {
+  if (vertOrHoriz === "horizontal") {
+    for (let i = columns; i < columns + ship.length; i++) {
+      board === PLAYER_BOARD
+        ? PLAYER_DIVS[rows * LENGTH + i].classList.add(
+            "player-ship",
+            "horizontal"
+          )
+        : CPU_DIVS[rows * LENGTH + i].classList.add("cpu-ship", "horizontal");
+      board === PLAYER_BOARD
+        ? (PLAYER_DIVS[rows * LENGTH + i].id = `${ship.key}`)
+        : (CPU_DIVS[rows * LENGTH + i].id = `${ship.key}`);
+      ship.firstIndex = `${rows}${columns}`;
+      board[rows][i] = 1;
+    }
+  } else {
+    for (let i = rows; i < rows + ship.length; i++) {
+      board === PLAYER_BOARD
+        ? PLAYER_DIVS[i * LENGTH + columns].classList.add(
+            "player-ship",
+            "vertical"
+          )
+        : CPU_DIVS[i * LENGTH + columns].classList.add("cpu-ship", "vertical");
+      board === PLAYER_BOARD
+        ? (PLAYER_DIVS[i * LENGTH + columns].id = `${ship.key}`)
+        : (CPU_DIVS[i * LENGTH + columns].id = `${ship.key}`);
+      ship.firstIndex = `${rows}${columns}`;
+      board[i][columns] = 1;
+    }
+  }
+}
+
+//Handling the player clicking the CPU board to fire
+async function handleMove(evt) {
+  let clickedDiv = evt.target;
+  if (
+    shipCounter <= 6 ||
+    clickedDiv.classList.contains("miss") ||
+    clickedDiv.classList.contains("hit-ship") ||
+    clickedDiv.classList.contains("sunken-ship") ||
+    turn === -1 ||
+    winner !== null
+  )
+    return;
+
+  let divIdx = CPU_DIVS.indexOf(clickedDiv);
+  let rows = parseInt(divIdx.toString().charAt(0));
+  let columns = parseInt(divIdx.toString().charAt(1));
+  let currentShip;
+  if (isNaN(columns)) {
+    columns = rows;
+    rows = 0;
+  }
+  if (CPU_BOARD[rows][columns] === 0) CPU_DIVS[divIdx].classList.add("miss");
+  if (clickedDiv.classList.contains("cpu-ship")) {
```

Comment:
Since we have the row/col indices here, why not check the board instead of the classes? Something like `if(CPU_BOARD[rows][columns] === 1)` then we know we're hitting a ship part that hasn't been hit already. This would also remove the need for the if statement on line 334. You could just do `currentShip.health--;` without any check.